engine/src/main/java/org/terasology/world/internal/WorldProviderCoreImpl.java

```diff
@@ -201,9 +200,9 @@ public Block setBlock(Vector3ic worldPos, Block type) {
             Vector3i blockPos = ChunkMath.calcBlockPos(JomlUtil.from(worldPos));
             Block oldBlockType = chunk.setBlock(blockPos, type);
             if (oldBlockType != type) {
-                BlockChange oldChange = blockChanges.get(worldPos);
```

Comment:
Note:
This was an error in the previous joml-migration commit. `blockChanges` uses TeraMath-vectors as keys, but here a JOML-vector is being used to find an element.
The TeraMath- and JOML-vector classes are not hash-equal.